### PR TITLE
[BasicUI] input widget, support html escape characters in input

### PR DESCRIPTION
**Pull request by mherwege**
Closes https://github.com/openhab/openhab-webui/issues/2675

HTML escape characters in input widgets were not properly supported, leading to an escaped character sequence under certain conditions.

### CODE REVIEW

File: bundles/org.openhab.ui.basic/web-src/smarthome.js [hunk 1]
```diff
@@ -2313,10 +2313,12 @@
...
 		_t.itemType = _t.parentNode.getAttribute(o.itemTypeAttribute);
 		_t.inputHint = _t.parentNode.getAttribute(o.inputHintAttribute);
-		_t.itemState = _t.parentNode.getAttribute(o.itemStateAttribute);
-		_t.unit = _t.parentNode.getAttribute(o.unitAttribute);
+		_t.itemState = smarthome.UI.unEscapeHtml(_t.parentNode.getAttribute(o.itemStateAttribute));
+		_t.unit = smarthome.UI.unEscapeHtml(_t.parentNode.getAttribute(o.unitAttribute));
 		_t.prefixField = _t.parentNode.parentNode.querySelector(".mdl-form__input-prefix");
 		_t.postfixField = _t.parentNode.parentNode.querySelector(".mdl-form__input-postfix");
+		_t.prefix = _t.prefixField ? smarthome.UI.unEscapeHtml(_t.prefixField.innerHTML) : null;
+		_t.postfix = _t.prefixField ? smarthome.UI.unEscapeHtml(_t.postfixField.innerHTML) : null;
```
Comment (lolodomo) [inline bundles/org.openhab.ui.basic/web-src/smarthome.js:2321]: Shout be _t.postfixField as condition

File: bundles/org.openhab.ui.basic/web-src/smarthome.js [hunk 2]
```diff
@@ -2372,10 +2374,8 @@
 		}
 
 		function cleanValue(value) {
-			var prefix = _t.prefixField !== null ? _t.prefixField.innerHTML : "";
-			var postfix = _t.postfixField !== null ? _t.postfixField.innerHTML : "";
-			var newValue = value.startsWith(prefix) ? value.substr(prefix.length) : value;
-			newValue = value.endsWith(postfix) ? newValue.substr(0, newValue.lastIndexOf(postfix)) : newValue;
+			var newValue = (_t.prefix && value.startsWith(_t.prefix)) ? value.substr(_t.prefix.length) : value;
+			newValue = (_t.postfix && value.endsWith(_t.postfix)) ? newValue.substr(0, newValue.lastIndexOf(_t.postfix)) : newValue;
```
Comment (lolodomo) [inline bundles/org.openhab.ui.basic/web-src/smarthome.js:2378]: Is condition `_t.prefix` equivalent to `_t.prefix !== null` ? I am just not 100% sure.

File: bundles/org.openhab.ui.basic/web-src/smarthome.js [hunk 3]
```diff
@@ -2772,6 +2772,12 @@
 			return escapedText;
 		};
 
+		_t.unEscapeHtml = function(html) {
+			var text = document.createElement("textarea");
+			text.innerHTML = html;
+			return text.value;
```
Comment (lolodomo) [inline bundles/org.openhab.ui.basic/web-src/smarthome.js:2778]: Is it a clean way to do ?
Doesn't this new element appear in the page after creation ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.ui.basic/web-src/smarthome.js (hunk 3 above)
Comment (mherwege) [reply bundles/org.openhab.ui.basic/web-src/smarthome.js:2778]: No, it doesn’t. This is the most commended way I could find.

File: bundles/org.openhab.ui.basic/web-src/smarthome.js (hunk 2 above)
Comment (mherwege) [reply bundles/org.openhab.ui.basic/web-src/smarthome.js:2378]: null, undefined or empty will all produce false, which is OK. This avoids issues with the following calls on it.

File: bundles/org.openhab.ui.basic/web-src/smarthome.js (hunk 1 above)
Comment (mherwege) [reply bundles/org.openhab.ui.basic/web-src/smarthome.js]: Thanks, fixed.

File: bundles/org.openhab.ui.basic/web-src/smarthome.js (hunk 3 above)
Comment (lolodomo) [reply bundles/org.openhab.ui.basic/web-src/smarthome.js:2779]: In the linked URL, someone proposed to remove the created element. So maybe something like that is even better ?
```
var text = document.createElement("textarea"),
    decodedHtml;;
text.innerHTML = html;
decodedHtml = text.value;
text.remove();
return decodedHtml;
```

Comment (mherwege) [reply bundles/org.openhab.ui.basic/web-src/smarthome.js:2779]: I also read that it is garbage collected anyway as no reference is held to it after returning from the function, and that remove is not available in all versions of js. The last point is probably not an issue anymore with versions currently used in browsers, but I don’t see why risking it.

**Pull request merged**